src/test/scheduler/scheduler_test.cpp

```diff
@@ -164,23 +166,109 @@ TEST_F(SchedulerTest, DiamondDependenciesWithScheduler) {
 }
 
 TEST_F(SchedulerTest, LinearDependenciesWithoutScheduler) {
+  CurrentScheduler::set(nullptr);
```

Comment:
Why is this necessary now and wasn't before. If you need to do it for every test, consider doing it in  SchedulerTest::SetUp().